### ISSUE TITLE
Can't apply rules to `create` action without affecting `new` because of aliasing

### ISSUE THREAD

**Issue by durierem** [CanCanCommunity/cancancan#832]
### Steps to reproduce

I have a resource for which I would want to have different rules for `#new` and `#create`. But when I do:
```rb
can :new, Foo # renders a form
can :create, Foo { |foo| <some condition on foo> } # some options for foo may not be authorized
```
Since `create` is an alias for `new` and `create` I can't have seperate rules for `create` only without affecting `new`

Can I opt-out of aliasing somehow?

### Expected behavior
N/A

### Actual behavior
N/A

### System configuration
**Rails version**: 6.1.7

**Ruby version**: 3.2.0

**CanCanCan version**: 3.5.0

**Comment by chmich** [comment]
similar case with `:update` and `:edit`, see [Stackoverflow issue](https://stackoverflow.com/questions/77073767/prohibit-update-in-cancan-conflicting-with-edit)

**Comment by durierem** [comment]
Redefining `can :edit` afterwards works but is very impractical. In my case, the original `can` definition is not straightforward and would require copy-pasting and keep in sync multiple rules between multiple files.

I don't understand how these aliases have been thought through without realizing that it clashes with the very same action it tries to alias. The reasoning is given as:

> This will be very convenient when we will authorize the Rails Controller actions.

Which, to my understanding, doesn't change anything apart from the fact that I can write `can :update` instead of `can [:edit, :update]`. When authorizing actions, we still check for the individual action name `can?(:edit, foo)` or `can?(:update, foo)`. May I have missed something?

**Comment by chmich** [comment]
Same like in my case: The first `can ...` has a lot of logic that i dont want to touch! And i have the feeling that the `can` is NOT following the docs but the `cannot` does. And that seems to be the problem. I have a idea of a workaround and will test it tomorrow.

**Comment by mameier** [comment]
For me, the aliases are very reasonable, even with the clash of names of `:create` and `:update`.

My understanding is:  
* `:new` shows the form to `:create` an object.
* `:edit`shows the form to `:update` an object.

so there is no reason to allow a `:new` without also allowing `:create` and no reason to allow `:edit` without also allowing `:update`.

All case I saw that seem to require such a decoupling were misuse of `:edit` to actually `:show` an object or something similar.

Use dedicated actions.  
You can use partials to combine the common parts of the view, but use dedicated controller actions.  
You are free to define additional actions to the standard CRUD actions and define abilities on these extra actions.

**Comment by chmich** [comment]
The workaround that i meant - and sorry - it would be a silly workaround - was idea to check at the end of ability.rb if a `can :edit ...` is set, then do the `cannot ...` and the set the `can :edit ...` back. Something like:

```
can_edit_project = can?(:edit, Project)
cannot(:update, Project, any-conditions)
can(:edit, Project) if can_edit_project
```

But that all is not working. The first step fails. I found no way to check at the bottom of a ability.rb if above a privilege is set or not. Here i am struggling now.

From my view: Aliases on`can` are working fine but `cannot` should be excluded from all aliases and this should be mentioned on docs.

**Comment by durierem** [comment]
@mameier Your point is completely valid... in an ideal world.

Also, organizing my app to satisfy CanCanCan's assumptions feels off. It should be the other way around.

**Comment by chmich** [comment]
Yes. Its really better to use dedicated controller actions like @mameier says.

But for the `cannot` action, which is rarely used and only in very specific places, I see aliases in the wrong place.

**Comment by chmich** [comment]
In my example, the customer did not want to have an edit and a show button separately. So if you really want to have "dedicated actions" at this point, that means separate routes (edit/show), separate request tests, separate route tests, and separate controller actions. The controller action needs to provide instance variables. And in the index view you would need a condition: If update allowed, then take the edit route helper, otherwise take the show route helper. For each view in the entire application

From a concise naming point of view, you have to separate: edit is edit and show is show. The same app has a "backoffice" with edit and show actions, and yes, of course it is cleaner.

But from a functional point of view, it is a lot of code for nothing else than always rendering the same partial with a form that is disabled in one case and enabled in the other case. 

My example application was taken from another developer. In short, I'd say: @mameier's statement should be a general recommendation, and in case of doubt: be more conciese. But is should not be a requirement.

**Comment by durierem** [comment]
Imagine a form for which some controls are locked because of insufficient permissions or a paid feature you don't have access to. In this case, it seems perfectly logical to me to have the `update` action be more restrictive than the `edit` action.

**Comment by newfylox** [comment]
@mameier 

> so there is no reason to allow a :new without also allowing :create and no reason to allow :edit without also allowing :update.

Of course there is a reason in a `belongs_to` and/or `has_many` associations and having multiple roles/permissions.

In my case, I want to create a `Member` that `belongs_to` a plan on a specific condition
```ruby
can [:create], Member, plan: { condition: false }
```
The user has many roles and permissions that let him create members with a dropdown of multiple plans. I'm not gonna create a `MembersController` for each different plan only for having different `[:new, :create]`.

I want a single `new` action as I have a single `create` action so that `Cancancan` will handle the rest

```ruby
def new
  @member = Member.new
end
```

**Comment by chmich** [comment]
Suggestion:

- `can_exact` function which works similar to `can`, but without aliases
- `cannot` should no longer follow aliases

**Comment by foton** [comment]
Or just add some way to disable, remove aliases.
